src/firewall/core/io/zone.py

```diff
@@ -245,11 +245,19 @@ def check_name(self, name):
         elif name.count('/') > 1:
             raise FirewallError(errors.INVALID_NAME,
                                 "more than one '/' in '%s'" % name)
-        elif len(name) > max_zone_name_len():
-            raise FirewallError(
-                errors.INVALID_NAME,
-                "'%s' has %d chars, max is %d" % (name, len(name),
-                                                  max_zone_name_len()))
+        else:
+            if "/" in name:
```

Comment:
AFAICS, this change is unnecessary since you added no_check_name  to zone_reader() for the combined case. We do check_name() for the combined_zone with the modified name.

Reply:
If memory serves me right, that part i needed for check_name(...) calls from firewalld-cmd, but some time has passed since the merge request...

Reply:
Alright. I'll try to create a test case for this #89 to verify. I hope to get to it soon, but I have some other things on my plate first.